extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala

```diff
@@ -61,6 +61,43 @@ abstract class RangerSparkExtensionSuite extends AnyFunSuite
     s"Permission denied: user [$user] does not have [$privilege] privilege on [$resource]"
   }
 
+  test("[KYUUBI #3276] InsertIntoHadoopFsRelationCommand: Write tables should also be checked ") {
+    val db = "db"
+    val createDB = s"CREATE DATABASE IF NOT EXISTS $db"
+    val table = "employee"
+    val createTable = s"CREATE TABLE IF NOT EXISTS $db.$table (id string) USING PARQUET"
+    val insert = s"INSERT INTO $db.$table values('id1')"
+
+    try {
+      doAs(
+        "admin", {
+          sql(createDB)
+          sql(createTable)
+
+          // Command may be executed early, so it should be CommandResult, however,
+          // this does not affect the original `Command` authentication, this is due to
+          // `eagerlyExecuteCommands` first plans the original command, this includes the
+          // Optimizer phase (i.e., authenticates the original command as well).
+          val optimizedPlan = sql(insert).queryExecution.optimizedPlan
+          assert(optimizedPlan.isInstanceOf[CommandResult])
+          assert(optimizedPlan.asInstanceOf[CommandResult].commandLogicalPlan.nodeName
+            == "InsertIntoHadoopFsRelationCommand")
+        })
+
+      doAs(
+        "yi", {
+          val e = intercept[RuntimeException](sql(insert).count())
+          assert(e.getMessage === errorMessage("update", s"$db/$table"))
+        })
+    } finally {
```

Comment:
not related to this PR, it's time to make a helper method for this try-finally block.

Reply:
Good idea, will submit new pr for this.